dclab/rtdc_dataset/ancillaries/af_image_contour.py

```diff
@@ -8,20 +9,13 @@ def compute_contour(mm):
     return cont
 
 
-def compute_bright_avg(mm):
-    bavg = features.bright.get_bright(mask=mm["mask"],
-                                      image=mm["image"],
-                                      ret_data="avg",
-                                      )
-    return bavg
-
-
-def compute_bright_sd(mm):
-    bstd = features.bright.get_bright(mask=mm["mask"],
-                                      image=mm["image"],
-                                      ret_data="sd",
-                                      )
-    return bstd
+@calltracker
```

Comment:
Yes, there is and I'm trying to get it to work with our package. The tricky thing is that `compute_bright` is not called within `AncillaryFeature`. It is called via `AncillaryFeature.compute` with the `AncillaryFeature.method` attribute. What I'm trying to figure out is: at what stage is "this has run once" sufficient. Presumably, checking if  `AncillaryFeature.compute` has been run once is not sufficient, as this is expected behaviour. Past that, I'm having trouble trying to mock the `compute_bright` function directly within the `AncillaryFeature` class. But that might not give us anything of worth either, because we know it will only be called once when we call it once.

Additionally, mocking `AncillaryFeature.compute` does not allow us to check whether the input `rtdc_dataset` has been populated with both `bright` features.

Example of mock and monkeypatch:

```python

from unittest.mock import Mock
import dclab.rtdc_dataset.ancillaries as ancill

def test_af_is_compute_called_only_once(monkeypatch):
    ds = dclab.new_dataset(retrieve_data("rtdc_data_traces_video_bright.zip"))

    mock_run = Mock()
    monkeypatch.setattr(
        'dclab.rtdc_dataset.ancillaries.AncillaryFeature.compute',
        mock_run)

    a = ancill.AncillaryFeature(
        feature_name="bright_sd",
        method=ancill.af_image_contour.compute_bright,
        req_features=["image", "mask"])
    _ = a.compute(ds)

    mock_run.assert_called_once()
```

There is always the option of taking all the `@calltracker` stuff I've written and putting it into the test function, along with some fake Ancillary Features (as you did above in your requested test). This would remove the issue of all of that test code making its way into the dclab package.

Reply:
In this example you monkeypatch `AncillaryFeature.compute`. If you monkeypatched `dclab.features.bright.get_bright` with a simple wrapper function (that calls the original function), this would be what we need I think.